src/backend/aspen/workflows/nextclade/prep_samples.py

```diff
@@ -18,44 +35,134 @@
+    - nextclade_dataset_dir: Dir to save the Nextclade dataset we download.
+    - nextclade_tag_filename: Name of file that Nextclade uses to tag datasets.
+    - job_info_fh: Write out info about job for later use in workflow
+
+    TODO (Vince): It would be nice if this process batched up pulling the FASTA
+    data over the samples. As it is, if a lot of samples need running, it just
+    grabs all the sequence data in one big go and writes it to a very large
+    FASTA. We haven't seen this approach cause issues so far -- it's how the
+    Pangolin job has been doing it when refreshing stale samples -- but it
+    could cause troubles as our sample count continues to grow and/or we are
+    dealing with much larger genomes. Long-term, it might be better to grab
+    chunks of samples from the DB, like 1k-10k at a time, so we don't have a
+    single, huge query result coming in from the DB.
     """
-    sample_ids: list[int] = [int(id_line) for id_line in sample_ids_fh]
     interface: SqlAlchemyInterface = init_db(get_db_uri(Config()))
     with session_scope(interface) as session:
         print(f"Getting pathogen data for {pathogen_slug}")
         target_pathogen_query: Pathogen = sa.select(Pathogen).filter(
             Pathogen.slug == pathogen_slug
         )
-        target_pathogen = session.execute(target_pathogen_query).scalars().one()
-        save_pathogen_info(pathogen_info_fh, target_pathogen)
+        target_pathogen: Pathogen = (
+            session.execute(target_pathogen_query).scalars().one()
+        )
+
+        # This downloads the dataset info to disk for later use by Nextclade.
+        nextclade_dataset_info = download_nextclade_dataset(
+            target_pathogen.nextclade_dataset_name,
+            nextclade_dataset_dir,
+            nextclade_tag_filename,
+        )
+
+        # Figure out which samples we need to run Nextclade on
+        sample_ids: list[int] = []
+        if run_type == RunType.SPECIFIED_IDS_ONLY:
+            # In case of "empty" sample IDs file, it still contains a newline
+            # because of how we build file upstream. Easiest way to handle
+            # empty case is to just ignore lines of only "\n" newline.
+            sample_ids = [int(id_line) for id_line in sample_ids_fh if id_line != "\n"]
+        elif run_type == RunType.REFRESH_STALE:
+            sample_ids = get_sample_ids_to_refresh(
+                session,
+                target_pathogen,
+                nextclade_dataset_info["name"],
+                nextclade_dataset_info["accession"],
+                nextclade_dataset_info["tag"],
+            )
+        elif run_type == RunType.FORCE_ALL:
+            sample_ids = get_all_sample_ids_for_pathogen(
+                session,
+                target_pathogen,
+            )
+
+        # We should abandon this workflow run if there are no samples to run.
+        if sample_ids == []:
+            if run_type == RunType.REFRESH_STALE:
```

Comment:
```suggestion
            if run_type != RunType.SPECIFIED_IDS_ONLY:
```
It's possible we won't have any samples in our system for a given pathogen, and we probably don't want to spam alerts if that's the case

Reply:
I agree generally, but right now the only other run type that would kick off an alert would be a `force-all` -- which is currently just for manual firing by an engineer in case something weird happens -- so I think it's reasonable for the alert to happen then. If we tried to trigger a `force-all` run for a pathogen with no samples at all, that seems like a usage mistake we would want to be notified about, not something quietly logged.

Reply:
An engineer running this manually will see the "no samples found" message and be able to decide what to do, while raising alerts when we have no samples is going to make it harder to automate this force-all job.

Reply:
I guess that I was imagining the force-all job to never be automated, manual only, so it wouldn't get too spammy. If an engineer is watching the logs, yeah, I agree, the alert is annoying. But I was thinking fail loud was better in case the eng forgets to monitor after kicking off a force-all.